pallets/hsm/src/lib.rs

```diff
@@ -827,18 +865,27 @@ pub mod pallet {
 
 			let collateral_info = Self::collaterals(collateral_asset_id).ok_or(Error::<T>::AssetNotApproved)?;
 
-			let (arb_direction, flash_loan_amount) = if let Some(arb_amount) = flash_amount {
-				ensure!(arb_amount > 0, Error::<T>::NoArbitrageOpportunity);
-				// if provided, we know what to do, but need to verify the size is ok
-				let pool_state = Self::get_stablepool_state(collateral_info.pool_id)?;
-				ensure!(
-					Self::check_trade_size(collateral_asset_id, &collateral_info, &pool_state, arb_amount),
-					Error::<T>::NoArbitrageOpportunity
-				);
-
-				(ARBITRAGE_DIRECTION_BUY, arb_amount)
-			} else {
-				Self::find_arbitrage_opportunity(collateral_asset_id).ok_or(Error::<T>::NoArbitrageOpportunity)?
+			let (arb_direction, flash_loan_amount) = match arbitrage {
+				Some(Arbitrage::HollarOut(arb_amount)) => {
+					ensure!(arb_amount > 0, Error::<T>::NoArbitrageOpportunity);
+					// if provided, we know what to do, but need to verify the size is ok
+					let pool_state = Self::get_stablepool_state(collateral_info.pool_id)?;
+					ensure!(
+						Self::check_trade_size(collateral_asset_id, &collateral_info, &pool_state, arb_amount),
+						Error::<T>::NoArbitrageOpportunity
+					);
+					Arbitrage::HollarOut(arb_amount).into()
+				}
+				Some(Arbitrage::HollarIn(_)) => {
```

Comment:
what is the reason that the arb amount is ignored here?

Reply:
there is a note just under it.
this would require to verify the amount in some way. 

i'd like to keep the original behaviour for now, hence it is not used. but using this approach will be clear idea what arb is executed by looking at the extrinisc parameters - whereas before it wasnt very clear.